### PR TITLE
Simplified isNumber function

### LINKED ISSUE #5675: Suggestion:  include an error message when drawing shapes with `NaN` values.
### Increasing Access

If the user tries to draw a shape where any of the arguments are `NaN` it does not draw anything.  I would expect that it would log an error using the friendly error system so that the user knows why their code didn't do what they intended.  This would make it much easier for beginners to debug their code.

### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [X] Friendly error system
- [ ] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

### Feature enhancement details

Initially I assumed that p5.js was somehow catching or avoiding a DOM error.  Unfortunately it seems like the canvas API itself does not give any errors or warnings when encountering `NaN` values (SVG does ☹️).  So p5.js would need to check and validate arguments on its own.

We would want to include an `NaN` check in all shape functions such as `rect()`, `line()`, etc.

Here is an example code but it's very uninteresting.  It doesn't do anything.
```
function setup() {
  createCanvas(400, 400);
  vanillaJs();
}

function draw() {
  background(220);
  fill(0);
  const myVar = undefined + 1;
  rect(0, myVar, 100, 100);
  line(100, 100, myVar, 100);
}

function vanillaJs() {
  const canvas = document.createElement('canvas');
  canvas.width = 400;
  canvas.height = 400;
  document.body.appendChild(canvas);
  const ctx = canvas.getContext('2d');
  ctx.fillStyle = 'red';
  const myVar = undefined + 1;
  ctx.fillRect(0, myVar, 100, 100);
}
```

### PR DESCRIPTION
**Pull request by anpanring**
Resolves #5675

Changes: Going off of discussion surrounding Issue #5675 and PR #5846, this simplifies the isNumber parameter validation function using parseFloat and isNaN to return false when users use null, NaN, or an empty string (w/ or without spaces) as a parameter when drawing shapes. It still accepts Infinity for special circumstances.

Screenshots of the change:
Screenshot 2023-03-31 at 1 19 58 PM

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by limzykenneth** [comment]
This implementation does not take into account some edge case behaviour of `parseFloat`, specifically `parseFloat` can parse a string as long as it starts with a number so `parseFloat("100abcd")` will get parsed into `100` and the whole implementation here will regard `"100abcd"` as a number when it is not. There are also number base annotated strings that cannot be handled by `parseFloat` as well but that is less of an issue.

File: test/unit/core/error_helpers.js [hunk 1]
```diff
@@ -177,6 +177,28 @@ suite('Error Helpers', function() {
...
+    testUnMinified('line: infinite value given', function() {
+      let err = assert.throws(function() {
+        p5._validateParameters('line', [1, 2, 4, Infinity]);
+      }, p5.ValidationError);
+      assert.strictEqual(
+        err.type,
+        'WRONG_TYPE',
+        'ValidationError type is correct'
+      );
+    });
```
Comment (limzykenneth) [inline test/unit/core/error_helpers.js:201]: CI is failing on this test case, I don't think we need to cover for `Infinity` as well since `Infinity` is a valid number that can be useful in certain circumstances (eg. `min()` `max()`)

Comment (anpanring) [reply test/unit/core/error_helpers.js]: My bad, that makes sense! I just pushed a commit to slightly change that test to check for NaN instead of Infinity since that should return a ValidationError and checking for NaN was originally the main objective.

**Comment by limzykenneth** [comment]
Looks good. Thanks!
